### DISCUSSION TITLE
windows无法使用SSH连接

### DISCUSSION THREAD

**Discussion by river7816** [natfrp/wiki#258] [使用问题]
我已经打开了Openssh服务并且以下两个服务均在运行状态。
<img width="531" alt="image" src="https://user-images.githubusercontent.com/114122443/211363920-51e01708-f395-4163-8e07-32af2001e7d5.png">
我按照要求添加了SSH隧道，设置如下，我不是特别理解为什么教程上的ip地址是127.0.0.1，但是我填写了我自己有限网卡的ipv4地址。
<img width="1121" alt="image" src="https://user-images.githubusercontent.com/114122443/211364474-c95f5b11-93ae-4d1b-b973-a58820f31a0f.png">
之后我通过在命令行里输入`ssh keygen`生成了private key：
<img width="672" alt="image" src="https://user-images.githubusercontent.com/114122443/211364965-f33be902-c6f4-4a21-86ea-7f4ec61f0900.png">
之后我在我的mac上通过Termius远程连接ssh
配置如下：
一开始我输入：cn-zz-bgp-3.natfrp.cloud连接，
<img width="347" alt="image" src="https://user-images.githubusercontent.com/114122443/211365788-b22cfeaa-0250-44a3-8f24-96548e35471e.png">
然后提示我输入用户名，我输入了我本地账户。
<img width="443" alt="image" src="https://user-images.githubusercontent.com/114122443/211365901-5909fe40-d056-4b88-84d2-45baa950dc57.png">
并且将ssh keygen的文件复制进去：
<img width="901" alt="image" src="https://user-images.githubusercontent.com/114122443/211367245-22553cc0-99ba-4aff-94cc-4ec69bf58e5d.png">
最总错误logs如下：
👤 Starting a new connection to: "cn-zz-bgp-3.natfrp.cloud" port "22"
⚙️ Starting address resolution of "cn-zz-bgp-3.natfrp.cloud"
⚙️ Address resolution finished
⚙️ Connecting to "198.18.1.128" port "22"
👤 Connection to "cn-zz-bgp-3.natfrp.cloud" established
⚙️ Starting SSH session
⚙️ Remote server: SSH-2.0-OpenSSH_7.4
⚙️ Agreed KEX algorithm: ecdh-sha2-nistp256
⚙️ Agreed Host Key algorithm: ssh-rsa
⚙️ Agreed server-to-client cipher: aes128-ctr MAC: hmac-sha2-256
⚙️ Agreed client-to-server cipher: aes128-ctr MAC: hmac-sha2-256
⚙️ Agreed client-to-server compression: none
⚙️ Agreed server-to-client compression: none
⚙️ Handshake finished
👤 Checking host key: 03:d3:63:27:68:4c:c4:ce:3d:c2:b7:c8:5b:db:ea:76
👤 Host "cn-zz-bgp-3.natfrp.cloud":"22" is known and matches
👤 Authenticating to "cn-zz-bgp-3.natfrp.cloud":"22" as "river7816@outlook.com"
❗ Disabling authentication using publickey: Private Key is empty
⚙️ Available client authentication methods: password,keyboard-interactive
⚙️ Authentication that can continue: publickey
⚙️ Starting SSH key selection
👤 An SSH key is selected
👤 Authenticating using publickey method
❗ Authentication failed (publickey)
⚙️ Partial success: no
⚙️ Authentication that can continue: publickey
😨 No more authentication methods to try

**Comment by fakeboboliu** [comment]
首先：请不要公开您的私钥，请立即重新生成密钥对；请仔细阅读文档并不要做出 “我不是特别理解为什么教程上要这样做但总之我就是要反着来” 的操作。

其次：端口输错了
![图片](https://user-images.githubusercontent.com/7552030/211369322-a829cd49-2f05-4342-9be0-c761de7d1e1f.png)

**Reply by river7816** [reply]
希望大佬能再指点一下。我已经更改为127.0.0.1并且将端口更改为45249，但仍然出现报错（由于我新建了一个线路，端口号变为32188）以下是报错内容
👤 Starting a new connection to: "cn-zz-bgp-3.natfrp.cloud" port "32188"
⚙️ Starting address resolution of "cn-zz-bgp-3.natfrp.cloud"
⚙️ Address resolution finished
⚙️ Connecting to "198.18.1.128" port "32188"
👤 Connection to "cn-zz-bgp-3.natfrp.cloud" established
⚙️ Starting SSH session
⚙️ Remote server: SSH-2.0-OpenSSH_for_Windows_8.1
⚙️ Agreed KEX algorithm: ecdh-sha2-nistp256
⚙️ Agreed Host Key algorithm: ssh-rsa
⚙️ Agreed server-to-client cipher: aes128-ctr MAC: hmac-sha2-256
⚙️ Agreed client-to-server cipher: aes128-ctr MAC: hmac-sha2-256
⚙️ Agreed client-to-server compression: none
⚙️ Agreed server-to-client compression: none
⚙️ Handshake finished
👤 Checking host key: 81:05:86:6e:d6:5c:9b:56:d9:ec:e1:b6:77:8b:d1:b6
👤 Host "cn-zz-bgp-3.natfrp.cloud":"32188" is known and matches
👤 Authenticating to "cn-zz-bgp-3.natfrp.cloud":"32188" as "river7816@outlook.com"
⚙️ Available client authentication methods: publickey,password,keyboard-interactive
⚙️ Authentication that can continue: publickey,keyboard-interactive
👤 Authenticating using publickey method
❗ Authentication failed (publickey)
⚙️ Partial success: no
⚙️ Authentication that can continue: publickey,keyboard-interactive
👤 Authenticating using keyboard-interactive method
❗ Authentication failed (keyboard-interactive)
⚙️ Partial success: no
⚙️ Authentication that can continue: publickey,keyboard-interactive
😨 No more authentication methods to try

**Reply by river7816** [reply]
问题已经解决，是authorized key的问题
